app/waf/src/controllers/monitor.controller.ts

```diff
@@ -54,7 +61,9 @@ export class MonitorController {
   async find(
     @param.query.object('filter', getFilterSchemaFor(Monitor)) filter?: Filter,
   ): Promise<CollectionResponse> {
-    const data = await this.monitorRepository.find(filter);
+    let tenantId = await this.reqCxt.get(WafBindingKeys.Request.KeyTenantId);
+    let tmpdata = await this.monitorRepository.find(filter);
+    const data = filterByTenantId<Monitor>(tmpdata, tenantId);
```

Comment:
Can we make that filtering happen in DB side, instead of loading everything in to memory and filter them? The data size would be huge, we will also need to implement pagination, in order to cap the size of loaded data.

Reply:
I guess it can be done by lower level ORM probably, but I am not sure it can be done by this.XXXRepository. I will have a look then.

Reply:
I am thinking If we combine user input filter and tenant id filter, which can filter objects that belong to the tenant at once, it may cause some security issues.

Reply:
@zhaoqin-github  “loading everything in to memory and filter them” "implement pagination, in order to cap the size of loaded data.", 
I think they may be not sure. It's not that all data loaded into memory first. 
For Paging, Repository may have supported this feature, I remembered. 

We can inline the tenant filter function into `CommonRepository.find`, is it?

Reply:
It is OK to implement the code in repository layer or controller layer, both work. The important thing is NOT to load the whole table from DB.